perception/autoware_scene_to_image_projector/src/autoware_scene_to_image_projector_node.cpp

```diff
@@ -0,0 +1,548 @@
+  latest_detected_objects_received_ = true;
+}
+
+void SceneToImageProjectorNode::tracked_objects_callback(
+  const autoware_perception_msgs::msg::TrackedObjects::ConstSharedPtr & msg)
+{
+  latest_tracked_objects_ = std::make_shared<autoware_perception_msgs::msg::TrackedObjects>(*msg);
+  latest_tracked_objects_received_ = true;
+}
+
+void SceneToImageProjectorNode::trajectory_callback(
+  const autoware_planning_msgs::msg::Trajectory::ConstSharedPtr & msg)
+{
+  latest_trajectory_ = std::make_shared<autoware_planning_msgs::msg::Trajectory>(*msg);
+  latest_trajectory_received_ = true;
+}
+
+void SceneToImageProjectorNode::path_callback(
+  const autoware_planning_msgs::msg::Path::ConstSharedPtr & msg)
+{
+  latest_path_ = std::make_shared<autoware_planning_msgs::msg::Path>(*msg);
+  latest_path_received_ = true;
+}
+
+void SceneToImageProjectorNode::image_callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & input_image_msg)
+{
+  if (camera_info_received_) {
+    cv_bridge::CvImagePtr cv_ptr = cv_bridge::toCvCopy(input_image_msg, "bgr8");
+    Eigen::Matrix4d projection = get_projection_matrix(*latest_camera_info_);
+
+    if (latest_detected_objects_received_) {
+      const auto self_transform = get_transform_anonymous(
+        tf_buffer_, latest_detected_objects_->header.frame_id, latest_camera_info_->header.frame_id,
+        latest_detected_objects_->header.stamp);
+
+      if (!self_transform) {
+        return;
+      }
+
+      /* transform to world coordinate */
+      autoware_perception_msgs::msg::DetectedObjects transformed_objects;
+      if (!object_recognition_utils::transformObjects(
+            *latest_detected_objects_, latest_camera_info_->header.frame_id, tf_buffer_,
+            transformed_objects)) {
+        return;
+      }
+
+      auto & objects = transformed_objects.objects;
+
+      std::sort(objects.begin(), objects.end(), [](const auto & obj1, const auto & obj2) {
+        return (
+          obj1.kinematics.pose_with_covariance.pose.position.x *
+              obj1.kinematics.pose_with_covariance.pose.position.x +
+            obj1.kinematics.pose_with_covariance.pose.position.y *
+              obj1.kinematics.pose_with_covariance.pose.position.y +
+            obj1.kinematics.pose_with_covariance.pose.position.z *
+              obj1.kinematics.pose_with_covariance.pose.position.z <
+          obj2.kinematics.pose_with_covariance.pose.position.x *
+              obj2.kinematics.pose_with_covariance.pose.position.x +
+            obj2.kinematics.pose_with_covariance.pose.position.y *
+              obj2.kinematics.pose_with_covariance.pose.position.y +
+            obj2.kinematics.pose_with_covariance.pose.position.z *
+              obj2.kinematics.pose_with_covariance.pose.position.z);
+      });
+
+      std::vector<std::vector<cv::Point2f>> previous_polygons;
+
+      for (const auto & object : objects) {
+        if (should_skip_object(object.classification.front().label)) {
```

Comment:
Sometimes object does not have classification, so I needed to check `object.classification.empty()`.
This happens when I was checking a local rosbag with logging simulator.
Could you check this too?